### PR TITLE
test/e2e: check if prometheus is reachable before querying for prom rules

### PR DESCRIPTION
**Pull request by paulfantom**
I glanced over our test failures from `e2e-aws-operator` and it looks like most of them are related to problems with connecting to prometheus because proxies didn't finish bootstrapping (returning 503 instead of 200). The change proposed here should fix this by ensuring connection to prometheus can be established before testing for rules.

/cc @openshift/openshift-team-monitoring 

* [ ] I added CHANGELOG entry for this change.
* [X] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: test/e2e/multi_namespace_test.go [hunk 1]
```diff
@@ -66,6 +68,24 @@ func TestMultinamespacePrometheusRule(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	var lastErr error
+	// wait for proxies bootstrap
+	err = wait.Poll(time.Second, 5*time.Minute, func() (bool, error) {
+		_, err := f.ThanosQuerierClient.Do("GET", "/status", nil)
```
Comment (simonpasquier) [inline test/e2e/multi_namespace_test.go:74]:
```suggestion
		_, err := f.ThanosQuerierClient.Do("GET", "/-/ready", nil)
```

**Comment by simonpasquier** [comment]
/lgtm
/hold
one small nit but nothing crucial

File: test/e2e/multi_namespace_test.go (hunk 1 above)
Comment (paulfantom) [reply test/e2e/multi_namespace_test.go]: Fixed

**Pull request merged**